src/transformers/models/whisper/generation_whisper.py

```diff
@@ -686,15 +687,25 @@ def generate(
             batch_size=cur_bsz,
             generation_config=generation_config,
         )
+        # 5bis speculative decoding: ensure the assistant model does only one call to generate and therefore returns decoder input token ids and eos token id
+        # we set a flag in the generation config to force the model to make only one call to generate and return the decoder input token ids and eos token id
+        if "assistant_model" in kwargs:
+            assistant_model = kwargs["assistant_model"]
+            assistant_model.generation_config.force_unique_generate_call = True
+
+        if hasattr(generation_config, "force_unique_generate_call"):
+            force_unique_generate_call = generation_config.force_unique_generate_call
+        elif hasattr(self.generation_config, "force_unique_generate_call"):
+            force_unique_generate_call = self.generation_config.force_unique_generate_call
+        else:
+            force_unique_generate_call = False
```

Comment:
What if `kwargs["force_unique_generate_call"]=True` ?

Reply:
I've added a `force_unique_generate_call` arg to generate 👌